### PR TITLE
Add string and byte serializer

### LINKED ISSUE #27: Support writing serialized protobuf objects
Hi @mostafa! 
How hard is it to add support writing to kafka _serialized_ protobuf objects? 

```javascript
export default function () {

  // build Protobuf object
  const protobufObject = new protobufs.ProtobufObject();
  protobufObject.setIdentifier(identifier);
  protobufObject.setEvent(eventSchema);

  // Write to kafka
  const error = produce(producer, signalEnvelope.serializeBinary()); // xk6 support only []map[string]string
  check(error, {
    'is sent': (err) => err == undefined,
  });
```
The Error is: 

```bash
ERRO[0001] could not convert 10,12,10,10,97,117,116,111,109,97,116,105,111,110,58,175,2,10,51,10,35,103,101,110,101,114,97,108,95,105,110,102,111,114,109,97,116,105,111,110,46,108,111,103,95,115,111,117,114,99,101,46,110,97,109,101,18,12,85,110,105,100,101,110,116,105,102,105,101,100,10,70,10,30,103,101,110,101,114,97,108,95,105,110,102,111,114,109,97,116,105,111,110,46,109,101,115,115,97,103,101,95,105,100,18,36,98,102,97,48,51,101,57,53,45,49,97,102,100,45,52,51,52,53,45,57,97,102,55,45,51,102,48,102,99,49,51,53,97,102,50,101,10,54,10,31,103,101,110,101,114,97,108,95,105,110,102,111,114,109,97,116,105,111,110,46,114,97,119,95,109,101,115,115,97,103,101,18,19,78,101,109,111,45,87,114,105,116,101,45,84,111,45,75,97,102,107,97,10,75,10,41,103,101,110,101,114,97,108,95,105,110,102,111,114,109,97,116,105,111,110,46,115,116,97,110,100,97,114,100,95,109,101,115,115,97,103,101,95,116,105,109,101,18,30,50,48,50,49,45,48,54,45,51,48,84,49,52,58,53,57,58,52,55,46,54,56,53,49,52,50,50,48,48,90,10,43,10,29,103,101,110,101,114,97,108,95,105,110,102,111,114,109,97,116,105,111,110,46,116,101,110,97,110,116,95,105,100,18,10,97,117,116,111,109,97,116,105,111,110 to []map[string]string
        at go.k6.io/k6/js/common.Bind.func1 (native)
```

### PR DESCRIPTION
**Pull request by thmshmm**
fixes #27 

Added support to serialize string and byte array keys/values. Also fixes producing Protobuf data when the data is converted to a byte array upfront.

### CODE REVIEW

File: serde.go
Comment (mostafa) [inline serde.go:18]: I think the default case should be a string.

**Review by mostafa — CHANGES_REQUESTED** [review]
Overall LGTM, but needs a small change. I have a question: might a user need to receive a byte array while consuming messages?

File: serde.go
Comment (thmshmm) [reply serde.go:18]: I can change it to string, just thought to keep it compatible with some already existing tests which expect the previous behavior. Shall I still replace it?

**Comment by thmshmm** [comment]
Yes, deserialization should be implemented as well. I can work on it if I have some time. For now the regular consumer can be used to "just" read some messages and check if they were written.

**Comment by mostafa** [comment]
@thmshmm Let's wait until you add that part, too, before merging the PR.

File: serde.go
Comment (thmshmm) [reply serde.go:18]: If e.g. Protobuf schemas will be supported in the future as well, then we should make it somehow required to specify the serializer not just providing a schema. Maybe the `produce` function should be dedicated to string serialization and the schema options should be removed. If a schema is used it should be provided using the `produceWithConfiguration` together with the appropriate serializer/deserializer configuration.

Comment (mostafa) [reply serde.go:18]: The `produceWithConfiguration` isn't exported to JS because it's not an exported function in Go. Still, having a single uniform interface, `produce`, would be better if we could have optional arguments so that we don't pass an empty string for the schema (and possibly no nulls for others).
I made an initial proposal and built it on top of @iamelevich's work (https://github.com/mostafa/xk6-kafka/pull/20) to pass an object to the writer and the reader interface for configuration, but it didn't turn out as expected. I also included many other options which were left out in the writer configuration in the struct but to no avail.

**Comment by thmshmm** [comment]
@mostafa can you review again pls? Added the deserialisation part as well.

**Comment by mostafa** [comment]
Hey @thmshmm,

Thanks again for your contribution. Have you tested this code? I tried it, and it fails on a check:
```bash
✗ payload is correct
↳  0% — ✓ 0 / ✗ 7
```
And the reason is that the returned data is a string, not a byte array.

**Comment by thmshmm** [comment]
It works for me `✓ payload is correct`. I also noticed this, but found no solution. Thats why I used `String.fromCharCode(...msgs[0].value.split(","))` to convert it back. I think this behaviour may come from the goja implementation that k6 uses.

**Comment by mostafa** [comment]
I used your script called `scripts/test_bytes.js`, which doesn't work. I will try again to see if I can find a solution for it.

**Comment by thmshmm** [comment]
Do you have any errors when creating the k6 bin?
`xk6 build --with github.com/thmshmm/xk6-kafka@1fd4880ca75cdf1c78930e419d192e3bf36397b82 --replace github.com/thmshmm/xk6-kafka@latest=github.com/mostafa/xk6-kafka@latest --output k6`

**Comment by mostafa** [comment]
@thmshmm Just found the problem! I'll make suggestions, so you can merge them right away into your PR.

File: scripts/test_bytes.js
Comment (mostafa) [inline scripts/test_bytes.js:9]:
```suggestion
import {
    writer,
    produceWithConfiguration,
    reader,
    consumeWithConfiguration,
    createTopic,
} from "k6/x/kafka"; // import kafka extension
```

File: scripts/test_bytes.js
Comment (mostafa) [inline scripts/test_bytes.js:62]:
```suggestion
        "payload is correct": (msgs) => String.fromCharCode(...msgs[0].value) === payload,
```

**Review by mostafa — CHANGES_REQUESTED** [review]
These are the changes you need to make to make the script work as expected. As far as I see, the implementation works as expected, but the script should be fixed.

**Comment by thmshmm** [comment]
@mostafa thanks for your support, I integrated all your corrections. But, for me line 62 only works like this
`"payload is correct": (msgs) => String.fromCharCode(...msgs[0].value.split(",")) === payload,` with the `.split(",")`. Do you have any idea?

**Review by mostafa — APPROVED** [review]
@thmshmm Thanks again for your contribution! I'll merge this and release a version after that.

As for your question, I tried these after `let messages = consumeWithConfiguration(consumer, 10, configuration);`:
```javascript
console.log(messages[0].value);
console.log(typeof messages[0].value);
console.log(typeof messages[0].value[0]);
```
and this is the k6 console output:
```bash
INFO[0000] 98,121,116,101,32,97,114,114,97,121,32,112,97,121,108,111,97,100  source=console
INFO[0000] object                                        source=console
INFO[0000] number                                        source=console
```
I also tried [String.fromCharCode](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode), and it only accepts a sequence (array?) of numbers and returns a `String`. Anything else isn't supported and will result in an empty string.

**Pull request merged**
